### DISCUSSION TITLE
Update paperless-ngx from 2.9.0 to 2.10.0 fails again (same issue 2.8.1 to 2.9.0 #2925)

### DISCUSSION THREAD

**Discussion by epcos71** [tteck/Proxmox#3234] [Help Needed]
As like the update issue from 2.8.1 to 2.9.0 it is now from 2.9.0 to 2.10.0.
The scripts works perfectly and says update is done, but when you refresh the browser it is 2.9.0 and the update option is gone. Looking into the system status, paperless says it is 2.10.0, but the new feature like the trash is not there.

I have debian bookworm and sqlite as db, so is not the db issue.

**Comment by Mellowlynx** [comment]
Sure it's not a cache issue? I did not have any problems with the updates and everything worked great after a refresh.

**Reply by epcos71** [reply]
Thats the very first thing I checked... still 2.9.0

**Comment by Citroene** [comment]
here no problem with updating 

![Schermafbeelding 2024-06-19 093818](https://github.com/tteck/Proxmox/assets/56793809/a380517e-ac70-45c6-864b-5350dba52693)

**Comment by smignon612** [comment]
update works fine here

**Comment by jsapede** [comment]
got 2.10 here 
![image](https://github.com/tteck/Proxmox/assets/66303110/4bbe1c9a-68a3-450a-847a-0cee8d6ecb2f)

but still 2.9 here 
![image](https://github.com/tteck/Proxmox/assets/66303110/5cfb3227-7737-4618-8e99-363e010531e7)

even after cache clean

**Comment by Rockhound53** [comment]
Hello, 

I just tried to update to v2.10.2, unfortunately I get the following error message:
![image](https://github.com/tteck/Proxmox/assets/98473599/1c975842-abfa-4348-8502-2916eda0333f)
Does anyone have a solution?

Best regards

**Reply by git5any** [reply]
@Rockhound53 : from what version did you start the update?
I also got this error a while ago. Sometimes the update was running, but after rebooting the Startpage of paperless-ngx showed the rotating ring and some other issues with consuming new pdf. This startet with v2.7.2. From time to time I tried a new update, but always the same behavior (before an update I always make a backup of the LXC and restore it if the update fails.)
What helped: I ran the exporter, build a complete new LXC with the tteck-script, copied the exported files in and ran the importer. Everything was there again and worked. I came from 2.7.2 and imported in a fresh 2.10.1. Just now I did the update to 2.10.2 and this one worked. (update in the console of the proxmox-LXC of paperless-ngx)
No I'm hoping that future updates work, like they did before 2.7.2 (in my case)

kind Regards

**Reply by Rockhound53** [reply]
Ich habe das Update von 2.10.1 gemacht. 

Das update ging wohl auch durch, evtl. hat die Fehlermeldung einen anderen Grund?

![image](https://github.com/tteck/Proxmox/assets/98473599/75a712fe-2b7e-4d65-9da7-8366b777e496)

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/issues/2925

**Reply by Rockhound53** [reply]
Hello, 

Unfortunately that didn't work for me.

I tried the following, but my database seems to already be on UTF-8:

https://www.bytehero.io/posts/2024/paperless-postgres-encoding-error/

![image](https://github.com/tteck/Proxmox/assets/98473599/b4972237-fa7d-449e-8e23-d2042f7ee80d)


What else can I try?

**Reply by Rockhound53** [reply]
Small update, this helped:

https://github.com/tteck/Proxmox/issues/2925#issuecomment-2107685988

Maybe this can be changed in the future script?

**Comment by MickLesk** [comment]
There were also a few problems with the new version. If the update does not work for you because the migrate fails, you can try the following. Everything went smoothly for me. 

Enter LXC Console
sudo -u postgres psql 
GRANT CREATE ON SCHEMA public TO **##DB_USER##**;
ALTER DATABASE **##DB_NAME##** OWNER TO **##DB_USER##**;


![image](https://github.com/tteck/Proxmox/assets/47820557/531e307d-94b6-4d2f-832b-941443d71e13)

**Reply by git5any** [reply]
after reading thru all these post I can confirm an update from version 2.7.2 to 2.10.2 is working after doing the steps described here:
https://www.bytehero.io/posts/2024/paperless-postgres-encoding-error/

There is just a small "error" in the description:  you also have to stop paperless-task-queue.service, otherwise you get reports that the database is still in use by someone.
as mentioned above another way was to run exporter from 2.7.2, creating a complete new LXC with the installscript and then use importer into the new and clean installation.

**Comment by Rockhound53** [comment]
Hello everyone, 

Unfortunately, after all the back and forth with the database, the classifier no longer learns?

<img width="850" alt="image" src="https://github.com/tteck/Proxmox/assets/98473599/6cd2735d-fbda-409d-abf5-ede6241b1a8d">

Does anyone have an idea what this could be?

thanks

**Reply by jsapede** [reply]
Check if the service is started ? Eventually delete the celery db and restant
